### ISSUE TITLE
KIC 2.0: handle stderrthreshold flag

### ISSUE THREAD

**Issue by rainest** [Kong/kubernetes-ingress-controller#1297]
Follow up from https://github.com/Kong/kubernetes-ingress-controller/pull/1294#pullrequestreview-655935956

Although the controller's logger doesn't make use of the `--stderrthreshold` flag, its client-go instance uses klog and does. However, Cobra doesn't play nicely with it; we need to figure out how to make it handle the klog flags properly.

**Comment by rainest** [comment]
Hmm, so maybe not as easy as I thought. https://github.com/Kong/kubernetes-ingress-controller/commit/852a8f50684ceb96df066580a4768513712f09c8 is based off a combination of
https://flowerinthenight.com/blog/2019/02/05/golang-cobra-klog
https://pkg.go.dev/github.com/spf13/pflag#readme-supporting-go-flags-when-using-pflag
https://pkg.go.dev/k8s.io/klog#readme-how-to-use-klog

But this results in a conflict:

```
$ ./bin/manager --help
 flag redefined: kubeconfig
panic:  flag redefined: kubeconfig
```
Removing our kubeconfig definition still overwrites Cobra's generated help with the klog flag set's help, and I'm not sure how we pull out the kubeconfig flag for use with controller-runtime init :(

**Comment by shaneutt** [comment]
After reading through this and given the complexities and cost-benefit considering that we've moved away from the previous logging mechanisms that employed this, I have opted to deprecate the flag but make it available for compatibility reasons in 2.0 #1512
